packages/camera/camera_android/test/android_camera_test.dart

```diff
@@ -696,221 +504,117 @@ void main() {
       await camera.setFlashMode(cameraId, FlashMode.auto);
       await camera.setFlashMode(cameraId, FlashMode.off);
 
       // Assert
-      expect(channel.log, <Matcher>[
-        isMethodCall('setFlashMode', arguments: <String, Object?>{
-          'cameraId': cameraId,
-          'mode': 'torch'
-        }),
-        isMethodCall('setFlashMode', arguments: <String, Object?>{
-          'cameraId': cameraId,
-          'mode': 'always'
-        }),
-        isMethodCall('setFlashMode',
-            arguments: <String, Object?>{'cameraId': cameraId, 'mode': 'auto'}),
-        isMethodCall('setFlashMode',
-            arguments: <String, Object?>{'cameraId': cameraId, 'mode': 'off'}),
-      ]);
+      verify(mockCameraApi.setFlashMode(cameraId, PlatformFlashMode.torch))
+          .called(1);
+      verify(mockCameraApi.setFlashMode(cameraId, PlatformFlashMode.always))
+          .called(1);
+      verify(mockCameraApi.setFlashMode(cameraId, PlatformFlashMode.auto))
+          .called(1);
+      verify(mockCameraApi.setFlashMode(cameraId, PlatformFlashMode.off))
+          .called(1);
     });
 
     test('Should set the exposure mode', () async {
       // Arrange
-      final MethodChannelMock channel = MethodChannelMock(
-        channelName: _channelName,
-        methods: <String, dynamic>{'setExposureMode': null},
-      );
-
       // Act
       await camera.setExposureMode(cameraId, ExposureMode.auto);
       await camera.setExposureMode(cameraId, ExposureMode.locked);
 
       // Assert
-      expect(channel.log, <Matcher>[
-        isMethodCall('setExposureMode',
-            arguments: <String, Object?>{'cameraId': cameraId, 'mode': 'auto'}),
-        isMethodCall('setExposureMode', arguments: <String, Object?>{
-          'cameraId': cameraId,
-          'mode': 'locked'
-        }),
-      ]);
+      verify(mockCameraApi.setExposureMode(cameraId, PlatformExposureMode.auto))
+          .called(1);
+      verify(mockCameraApi.setExposureMode(
+              cameraId, PlatformExposureMode.locked))
+          .called(1);
     });
 
     test('Should set the exposure point', () async {
       // Arrange
-      final MethodChannelMock channel = MethodChannelMock(
-        channelName: _channelName,
-        methods: <String, dynamic>{'setExposurePoint': null},
-      );
-
       // Act
       await camera.setExposurePoint(cameraId, const Point<double>(0.5, 0.5));
       await camera.setExposurePoint(cameraId, null);
 
       // Assert
-      expect(channel.log, <Matcher>[
-        isMethodCall('setExposurePoint', arguments: <String, Object?>{
-          'cameraId': cameraId,
-          'x': 0.5,
-          'y': 0.5,
-          'reset': false
-        }),
-        isMethodCall('setExposurePoint', arguments: <String, Object?>{
-          'cameraId': cameraId,
-          'x': null,
-          'y': null,
-          'reset': true
-        }),
-      ]);
+      verify(mockCameraApi.setExposurePoint(cameraId, argThat(isNotNull)))
```

Comment:
A lot of tests seem to have lost assertions about the arguments being correct. E.g., this should still be asserting that x and y are being passed the correct values (and x and y should be different so this would catch accidentally sending the wrong one for one of them; that's a latent issue with this particular test). Please audit to make sure that we're not losing coverage about the arguments being correct. In some cases we get that from mocks requiring specific arguments, but cases like here we need to capture the args and assert things about them.

Reply:
I've modified a few tests to use `predicate` to require that the arguments' fields match. Since pigeons don't implement equality, we can't just use the object itself as a matcher.